src/utils/bivariate/labelFormatters.ts

```diff
@@ -7,3 +8,30 @@ export const convertDirectionsArrayToLabel = (directions: string[][]) => {
   const [from = '', to = ''] = directions;
   return `${formatSentimentDirection(from)} → ${formatSentimentDirection(to)}`;
 };
+
+export const formatBivariateAxisLabel = (quotients: Axis['quotients']): string => {
+  if (!quotients) return '';
+  const [numerator, denominator] = quotients;
+
+  // no numerator unit - don't show units
+  if (isNilOrOther(numerator.unit.id)) {
+    return `${numerator.label} to ${denominator.label}`;
+  }
+  // numerator unit + denominator unit - one
+  if (denominator.name === 'one') {
+    return `${numerator.label} (${numerator.unit.shortName})`;
+  }
+  //  numerator unit + no denominator unit - show only numerator unit
+  if (isNilOrOther(denominator.unit.id)) {
+    return `${numerator.label} to ${denominator.label} (${numerator.unit.shortName})`;
+  }
+  // cases for both units
+  if (denominator.name === 'area_km2') {
+    return `${numerator.label} (${numerator.unit.shortName}/${denominator.unit.shortName})`;
+  }
+
+  return `${numerator.label} to ${denominator.label} (${numerator.unit.shortName}/${denominator.unit.shortName})`;
+};
+
+export const isNilOrOther = (unitId: string | null): boolean =>
+  !unitId || unitId === 'other';
```

Comment:
May be smth like this
I'm changing negative `isNilOrOther ` to avoid nesting negative statements
```suggestion

  // don't show units if there are none
  if (!hasUnits(nominator.unit.id)) {
    return `${nominator.label} to ${denominator.label}`;
  }
  
  if (denominator.name === 'one') {
    return `${nominator.label} (${nominator.unit.shortName})`;
  }
  // if no denominator unit - show only nominator unit
  if (!hasUnits(denominator.unit.id)) {
    return `${nominator.label} to ${denominator.label} (${nominator.unit.shortName})`;
  }
  // here we process cases where we have both units
  if (denominator.name === 'area_km2') {
    return `${nominator.label} (${nominator.unit.shortName}/${denominator.unit.shortName})`;
  }

  return `${nominator.label} to ${denominator.label} (${nominator.unit.shortName}/${denominator.unit.shortName})`;
};

const hasUnits = (unitId: string | null): boolean => Boolean(unitId && unitId !== 'other');
```

Reply:
hm, as far as I remember from clean code it's kinda a bad smell 
https://github.com/ryanmcdermott/clean-code-javascript#avoid-negative-conditionals
https://www.obsidiansecurity.com/blog/why-to-stop-writing-negative-code/

Reply:
Exactly, and `isNillOrOther` is a negative code. And in case you'll ever go with `!isNillOrOther` it would be even worse. 
`if (!condition)` is negative too, but much more convenient for developers to read because of `!` to reverse statement

Reply:
Ah, I thought it's more about usage with `!` in if statements, that it adds complexity.

Reply:
thanks for explanation! 👍